renter/renterutil/filesystem.go

```diff
@@ -460,11 +462,17 @@ func (fs *PseudoFS) Close() error {
 		}
 		delete(fs.files, fd)
 	}
+	var err *multierror.Error
 	for fd, d := range fs.dirs {
-		d.Close()
+		if e := d.Close(); e != nil {
+			err = multierror.Append(err, e)
+		}
```

Comment:
There's rarely any reason to care about `Close` failing on a directory, but I guess this should be handled. We can use a simple `[]string` and `errors.New` for this.

Reply:
Unfortunately, such a simple implementation with `[]string` and `errors.New` doesn't work because we cannot retrieve individual errors. As far as I know, go-multierror is one of the best options that allow us to check if a returned error containers some specific error.

Reply:
I can't think of any realistic circumstance in which you would need to programmatically handle the individual errors from closing directories. This is the sort of error that you would, *at most*, log somewhere and then manually inspect later. But in reality if you get an error when closing a directory, it's probably because your disk is failing and your log is gonna have a LOT of errors alerting you to that fact (assuming your system is usable at all).

Reply:
We cannot know what error happens in advance. It's always better to prepare for it.
Actually, if the error means the closing directory/file doesn't exist, the user might remove it and we can ignore/warn it, but if disk drive dies, we need to log it.

We need to debug our applications remotely. Any information is helpful.